bin/laminas

```diff
@@ -28,11 +26,5 @@ if (file_exists($a = __DIR__ . '/../../../autoload.php')) {
 // Set the main application directory as the current working directory
 chdir(dirname($a) . '/../');
 
-$container = ContainerResolver::resolve();
-$config = $container->get('config')['laminas-cli'] ?? [];
-
-$version = strstr(Versions::getVersion('laminas/laminas-cli'), '@', true);
-
-$application = new Application('laminas', $version);
-$application->setCommandLoader(new ContainerCommandLoader($container, $config['commands'] ?? []));
-$application->run();
+$applicationFactory = new ApplicationFactory();
+exit($applicationFactory(ContainerResolver::resolve())->run());
```

Comment:
```suggestion
$container = ContainerResolver::resolve();
exit($container->get(Application::class)->run());
```

Why not map the console application to the factory and then pull it from the container? The approach here feels a little backwards to how we do everything else.

Reply:
Because then we need expose Laminas\Cli module / config provider and register within the application.

Also I am not sure if I want to register it in the container, as it means user will be able to add decorator, change the application and change the behaviour we are trying to provide here.

I've marked factory as `@internal`.

Reply:
Maybe make a note in the `ApplicationFactory` docblock with your comments from above, so that we don't get a PR later suggesting putting it in the container. Otherwise, :ship: